src/main/java/com/mineboundteam/minebound/magic/PassiveSpellItem.java

```diff
@@ -26,7 +46,39 @@ protected static <T extends PassiveSpellItem> List<T> getEquippedSpellsOfType(Cl
                         spells.add((T) item.getItem());
                 }
             });
-
         return spells;
     }
+
+    @SuppressWarnings("unchecked")
+    protected static <T extends PassiveSpellItem> ItemStack getHighestEquippedSpellOfType(Class<T> type, List<ItemStack> spells) {
```

Comment:
I feel like this could be of type or from list, but it doesn't make much sense to be both. You could either send in the type, call getEquippedSpellItemsOfType() or just iterate through a list and return a PassiveSpellItem.

Reply:
Done.